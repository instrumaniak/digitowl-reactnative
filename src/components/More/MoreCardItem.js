import React from 'react'
import PropTypes from 'prop-types'
import {
  View,
  Text,
  StyleSheet
} from 'react-native'

import { Icon } from 'native-base'
import { colors } from '../../config'
import { human } from 'react-native-typography'

const MoreCardItem = ({
  iconText,
  title,
  description
}) => (
  <View style={styles.container}>
    { iconText &&
      <View style={styles.iconContainer}>
        <Text style={styles.iconText}>{iconText}</Text>
      </View>
    }
    <View style={styles.dataContainer}>
      <View>
        <Text style={styles.title}>{ title }</Text>
        { description && <Text style={styles.description}>{ description }</Text>}
      </View>
      <View>
        <Icon name='ios-arrow-forward' style={styles.icon}/>
      </View>
    </View>
  </View>
)

MoreCardItem.propTypes = {
  iconText: PropTypes.string,
  title: PropTypes.string,
  description: PropTypes.string
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'row',
  },
  iconContainer: {
    width: 50,
    height: 50,
    backgroundColor: colors.grey,
    borderRadius: 25,
    justifyContent: 'center',
    alignItems: 'center'
  },
  iconText: {
    color: colors.white,
    //fontWeight: 'bold',
    fontSize: 35
  },
  icon: {
    color: colors.grey1
  },
  dataContainer: {
    flex: 1,
    paddingLeft: 15,
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center'
  },

  // typography
  title: {
    ...human.bodyObject,
    color: colors.grey1,
    //fontSize: 18,
    //fontWeight: 'bold'
  },
  description: {
    ...human.subheadObject,
    color: colors.grey,
    fontSize: 16
  },
  value: {
    color: colors.grey1,
    fontSize: 18,
    fontWeight: 'bold'
  }
})

export default MoreCardItem
