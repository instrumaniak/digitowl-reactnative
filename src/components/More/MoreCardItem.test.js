import React from 'react'
import MoreCardItem from './MoreCardItem'
import renderer from 'react-test-renderer'

test('Renders correctly', () => {
  const tree = renderer.create(<MoreCardItem />).toJSON()
  expect(tree).toMatchSnapshot()
})
