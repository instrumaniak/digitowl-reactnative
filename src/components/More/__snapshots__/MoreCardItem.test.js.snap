// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingLeft": 15,
      }
    }
  >
    <View>
      <Text
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#5B6267",
            "fontFamily": "System",
            "fontSize": 17,
            "fontWeight": "400",
            "letterSpacing": -0.408,
            "lineHeight": 22,
          }
        }
      />
    </View>
    <View>
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Array [
              Object {
                "color": "#000",
                "fontSize": 30,
              },
              Object {
                "color": "#000",
                "fontSize": 30,
              },
              Object {
                "color": "#5B6267",
              },
            ],
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</View>
`;
